src/main/java/mSearch/filmlisten/WriteFilmlistJson.java

```diff
@@ -22,112 +22,166 @@
-                jg = jsonF.createGenerator(new File(datei), JsonEncoding.UTF8);
-            }
-            jg.useDefaultPrettyPrinter(); // enable indentation just to make debug/testing easier
-            jg.writeStartObject();
-            // Infos zur Filmliste
-            jg.writeArrayFieldStart(ListeFilme.FILMLISTE);
-            for (int i = 0; i < ListeFilme.MAX_ELEM; ++i) {
-                jg.writeString(listeFilme.metaDaten[i]);
-            }
-            jg.writeEndArray();
-            // Infos der Felder in der Filmliste
-            jg.writeArrayFieldStart(ListeFilme.FILMLISTE);
-            for (int i = 0; i < DatenFilm.JSON_NAMES.length; ++i) {
-                jg.writeString(DatenFilm.COLUMN_NAMES[DatenFilm.JSON_NAMES[i]]);
-            }
-            jg.writeEndArray();
-            //Filme schreiben
-            DatenFilm datenFilm;
-            Iterator<DatenFilm> iterator = listeFilme.iterator();
-            while (iterator.hasNext()) {
-                datenFilm = iterator.next();
-                datenFilm.arr[DatenFilm.FILM_NEU] = Boolean.toString(datenFilm.isNew()); // damit wirs beim nächsten Programmstart noch wissen
-
-                jg.writeArrayFieldStart(DatenFilm.TAG_JSON_LIST);
+
+            try (FileOutputStream fos = new FileOutputStream(datei);
+                 JsonGenerator jg = getJsonGenerator(fos)) {
+
+                jg.writeStartObject();
+                // Infos zur Filmliste
+                jg.writeArrayFieldStart(ListeFilme.FILMLISTE);
+                for (int i = 0; i < ListeFilme.MAX_ELEM; ++i) {
+                    jg.writeString(listeFilme.metaDaten[i]);
+                }
+                jg.writeEndArray();
+                // Infos der Felder in der Filmliste
+                jg.writeArrayFieldStart(ListeFilme.FILMLISTE);
                 for (int i = 0; i < DatenFilm.JSON_NAMES.length; ++i) {
-                    int m = DatenFilm.JSON_NAMES[i];
-                    if (m == DatenFilm.FILM_SENDER) {
-                        if (datenFilm.arr[m].equals(sender)) {
-                            jg.writeString("");
-                        } else {
-                            sender = datenFilm.arr[m];
-                            jg.writeString(datenFilm.arr[m]);
-                        }
-                    } else if (m == DatenFilm.FILM_THEMA) {
-                        if (datenFilm.arr[m].equals(thema)) {
-                            jg.writeString("");
+                    jg.writeString(DatenFilm.COLUMN_NAMES[DatenFilm.JSON_NAMES[i]]);
+                }
+                jg.writeEndArray();
+                //Filme schreiben
+                for (DatenFilm datenFilm : listeFilme) {
+                    datenFilm.arr[DatenFilm.FILM_NEU] = Boolean.toString(datenFilm.isNew()); // damit wirs beim nächsten Programmstart noch wissen
+
+                    jg.writeArrayFieldStart(DatenFilm.TAG_JSON_LIST);
+                    for (int i = 0; i < DatenFilm.JSON_NAMES.length; ++i) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS134)